apps/vaults/hooks/useSolverPortals.ts

```diff
@@ -28,77 +27,49 @@ export type TPortalsQuoteResult = {
 	error?: Error;
 };
 
-function useQuote(): [
-	TPortalsQuoteResult,
-	(request: TInitSolverArgs, shouldPreventErrorToast?: boolean) => Promise<TPortalEstimate | null>
-] {
-	const {toast} = yToast();
-	const {zapSlippage} = useYearn();
-	const [err, set_err] = useState<Error>();
-	const {getEstimate} = usePortalsApi();
-	const {safeChainID} = useChainID();
-
-	const getQuote = useCallback(async (
-		request: TInitSolverArgs,
-		shouldPreventErrorToast = false
-	): Promise<TPortalEstimate | null> => {
-		const params = {
-			sellToken: toAddress(request.inputToken.value),
-			sellAmount: toBigInt(request.inputAmount || 0).toString(),
-			buyToken: toAddress(request.outputToken.value),
-			slippagePercentage: String(zapSlippage / 100)
-		};
-
-		const canExecuteFetch = (
-			!(isZeroAddress(params.sellToken) || isZeroAddress(params.buyToken)) &&
-				toBigInt(request.inputAmount) !== 0n
-		);
-
-		if (!canExecuteFetch) {
-			return null;
+async function getQuote(
+	request: TInitSolverArgs,
+	safeChainID: number,
+	zapSlippage: number
+): Promise<{data: TPortalEstimate | undefined, error: Error | undefined}> {
+	const params = {
+		sellToken: toAddress(request.inputToken.value),
+		sellAmount: toBigInt(request.inputAmount || 0).toString(),
+		buyToken: toAddress(request.outputToken.value),
+		slippagePercentage: String(zapSlippage / 100)
+	};
+
+	if (isZeroAddress(params.sellToken)) {
+		return {data: undefined, error: new Error('Invalid sell token')};
+	}
+	if (isZeroAddress(params.buyToken)) {
+		return {data: undefined, error: new Error('Invalid buy token')};
+	}
+	if (toBigInt(params.sellAmount) === 0n) {
+		return {data: undefined, error: new Error('Invalid sell amount')};
+	}
```

Comment:
If `undefined` we don't need to explicitly set it
```suggestion
	if (isZeroAddress(params.sellToken)) {
		return {error: new Error('Invalid sell token')};
	}
	if (isZeroAddress(params.buyToken)) {
		return {error: new Error('Invalid buy token')};
	}
	if (toBigInt(params.sellAmount) === 0n) {
		return {error: new Error('Invalid sell amount')};
	}
```